tools/pyboard.py

```diff
@@ -491,7 +491,16 @@ def fs_cat(self, src, chunk_size=256):
         )
         self.exec_(cmd, data_consumer=stdout_write_bytes)
 
-    def fs_get(self, src, dest, chunk_size=256):
+    def fs_get(self, src, dest, chunk_size=256, progress_callback=None):
+        if progress_callback:
+            src_size = bytearray()
+            self.exec_(
+                "import os\nprint(os.stat('%s')[6])" % src,
+                data_consumer=lambda d: src_size.extend(d),
+            )
+            assert src_size.endswith(b"\r\n\x04")
+            src_size = int(src_size[:-3])
```

Comment:
I think this can be simplified to not use `data_consumer` (I see you copied this from below, and I'm not exactly sure why `data_consumer` is used below..).

Eg:
```python
src_size = int(self.exec_("..."))
```

Reply:
Yea that works and is a lot simpler.